kimchi/src/circuits/expr.rs

```diff
@@ -622,8 +622,45 @@ pub enum Expr<C, Col: ColTrait = Column> {
     IfFeature(FeatureFlag, Box<Expr<C, Col>>, Box<Expr<C, Col>>),
 }
 
-impl<C: Zero + One + Neg<Output = C> + PartialEq + Clone> Expr<C> {
-    fn apply_feature_flags_inner(&self, features: &FeatureFlags) -> (Expr<C>, bool) {
+impl<C, Col> Expr<C, Col>
+where
+    C: Zero + One + Neg<Output = C> + PartialEq + Clone,
+    Col: ColTrait,
+{
+    ///maps expressiosn over some columns to expressions over other columns
+    pub fn map_column<Col2, M>(self, map: &M) -> Expr<C, Col2>
+    where
+        Col2: ColTrait,
+        M: Fn(Col) -> Col2,
+    {
+        match self {
+            Expr::Constant(c) => Expr::Constant(c),
+            Expr::Cell(v) => {
+                let Variable { col, row } = v;
+                Expr::Cell(Variable { col: map(col), row })
+            }
+            Expr::Double(exp) => Expr::Double(Box::new((*exp).map_column(map))),
+            Expr::Square(exp) => Expr::Square(Box::new((*exp).map_column(map))),
+            Expr::BinOp(op, exp1, exp2) => {
+                let exp1 = exp1.map_column(map);
+                let exp2 = exp2.map_column(map);
+                Expr::BinOp(op, Box::new(exp1), Box::new(exp2))
+            }
+            Expr::VanishesOnLast4Rows => Expr::VanishesOnLast4Rows,
```

Comment:
Again, here this expression no longer exists in develop. Please get the most updated develop (or wait until we know what to do regarding `develop` vs `master`)